app/src/main/java/org/fossasia/openevent/adapters/LocationsListAdapter.java

```diff
@@ -105,18 +107,27 @@ public void moveItem(int fromPosition, int toPosition) {
         notifyItemMoved(fromPosition, toPosition);
     }
 
-    class Viewholder extends RecyclerView.ViewHolder {
+    public static interface AdapterOnClickHandler {
+        void onClick(int pos, Viewholder vh);
+    }
+
+    public static class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
         TextView name;
         TextView floor;
 
         public Viewholder(View itemView) {
             super(itemView);
             itemView.setClickable(true);
-
+            itemView.setOnClickListener(this);
             name = (TextView) itemView.findViewById(R.id.location_name);
             floor = (TextView) itemView.findViewById(R.id.location_floor);
 
         }
 
+        @Override
+        public void onClick(View view) {
+            clickHandler.onClick(getAdapterPosition(), this);
```

Comment:
no making it static does not allow it to have inner references so it is always good practice to use static inner classes unless you absolute need the outer reference. Please see: http://www.androiddesignpatterns.com/2013/01/inner-class-handler-memory-leak.html

Its for Handlers but applies generally to any Android code that has lifecycle issues like adapter, views, activities etc.